lib/events/forward.go

```diff
@@ -64,47 +66,101 @@ func NewForwarder(cfg ForwarderConfig) (*Forwarder, error) {
 type Forwarder struct {
 	ForwarderConfig
 	sessionLogger *DiskSessionLogger
+	lastChunk     *SessionChunk
+	eventIndex    int64
+	sync.Mutex
+	isClosed bool
 }
 
 // Closer releases connection and resources associated with log if any
 func (l *Forwarder) Close() error {
+	l.Lock()
+	defer l.Unlock()
+	if l.isClosed {
+		return nil
+	}
+	l.isClosed = true
```

Comment:
Shouldn't this flag be set only if Finalize below succeeds?

Reply:
not really, as we have attempted at least once and will not want to try again